camel-idea-plugin/src/main/java/org/apache/camel/idea/gutter/CamelRouteLineMarkerProvider.java

```diff
@@ -69,4 +93,116 @@ private CamelPreferenceService getCamelPreferenceService() {
+                if (xmlElement != null) {
+                    psiElements.add(xmlElement);
+                }
+            }
+            if (psiElement instanceof PsiLiteralExpression) {
+                PsiElement javaElement = findJavaElement(route, (PsiLiteralExpression) psiElement);
+                if (javaElement != null) {
+                    psiElements.add(javaElement);
+                }
+            }
+            return true;
+        }, getSearchScope(startElement.getProject()), componentName, UsageSearchContext.ANY, false);
+
+        return psiElements;
+    }
+
+    /**
+     * Further refine search in order to match the exact Java Camel route.
+     * Checks if the given {@link PsiElement} contains a 'to' method that points to the give route.
+     *
+     * @param route      the complete Camel route to search for
+     * @param psiElement the {@link PsiLiteralExpression} that might contain the complete route definition
+     * @return the {@link PsiElement} that contains the exact match of the Camel route, null if there is no exact match
+     */
+    private PsiElement findJavaElement(String route, PsiLiteralExpression psiElement) {
+        if (route.equals(psiElement.getValue())) {
+            //the method 'to' is a PsiIdentifier not a PsiMethodCallExpression because it's part of method invocation chain
+            PsiMethodCallExpression methodCall = PsiTreeUtil.getParentOfType(psiElement, PsiMethodCallExpression.class);
+            if (methodCall != null && "to".equals(methodCall.getMethodExpression().getReferenceName())) {
+                return psiElement;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Further refine search in order to match the exact XML Camel route.
+     *
+     * @param route      the complete Camel route to search for
+     * @param psiElement the {@link PsiElement} that might contain the complete route definition
+     * @return the {@link PsiElement} that contains the exact match of the Camel route
+     */
+    private PsiElement findXMLElement(String route, XmlToken psiElement) {
+        if (psiElement.getTokenType() == XmlElementType.XML_ATTRIBUTE_VALUE_TOKEN) {
+            if ("to".equals(PsiTreeUtil.getParentOfType(psiElement, XmlTag.class).getLocalName())) {
+                if (psiElement.getText().equals(route)) {
+                    return psiElement;
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Defines the search scope for the Camel routes.
+     * It consist of all the files from all the modules including the files from JAR files.
+     *
+     * @param project the current {@link Project}
+     * @return the search scope
+     */
+    private GlobalSearchScope getSearchScope(Project project) {
+        Module[] modules = ModuleManager.getInstance(project).getModules();
+
+        GlobalSearchScope searchScope = GlobalSearchScope.EMPTY_SCOPE;
```

Comment:
Are we not searching to far? I think you should only search in the project source code, eg `src/main/java` and maybe also `src/test/java` etc. Not in JARs and other libraries.